tests/suite/tracing_test.go

```diff
@@ -0,0 +1,270 @@
+		}
+
+		// wait for expected first line to show up
+		Eventually(findTraces, "1m", "5s").Should(BeTrue())
+
+		logs, err = resourceManager.GetPodLogs(collectorNamespace, collectorPodName, &core.PodLogOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(logs).To(ContainSubstring("http.method: Str(GET)"))
+		Expect(logs).To(ContainSubstring("http.target: Str(/hello)"))
+		Expect(logs).To(ContainSubstring("testkey1: Str(testval1)"))
+		Expect(logs).To(ContainSubstring("testkey2: Str(testval2)"))
+
+		// verify traces don't exist for other apps
+		Expect(logs).ToNot(ContainSubstring("http.target: Str(/world)"))
+		Expect(logs).ToNot(ContainSubstring("http.target: Str(/helloworld)"))
+	})
+
+	It("sends tracing spans for one policy attached to multiple routes", func() {
+		// install tracing configuration
+		traceFiles := []string{
+			"tracing/nginxproxy.yaml",
+			"tracing/policy-multiple.yaml",
+		}
+		Expect(resourceManager.ApplyFromFiles(traceFiles, ns.Name)).To(Succeed())
+		Expect(updateGatewayClass()).To(Succeed())
+
+		Eventually(
+			func() error {
+				return verifyGatewayClassResolvedRefs()
+			}).
+			WithTimeout(timeoutConfig.GetTimeout).
+			WithPolling(500 * time.Millisecond).
+			Should(Succeed())
+
+		Eventually(
+			func() error {
+				return verifyPolicyStatus()
+			}).
+			WithTimeout(timeoutConfig.GetTimeout).
+			WithPolling(500 * time.Millisecond).
+			Should(Succeed())
+
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		// send traffic and verify that traces exist for hello app
+		findTraces := func() bool {
+			sendTraceRequests(ctx, helloURL, 25)
+			sendTraceRequests(ctx, worldURL, 25)
+			sendTraceRequests(ctx, helloworldURL, 25)
+
+			logs, err := resourceManager.GetPodLogs(collectorNamespace, collectorPodName, &core.PodLogOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return strings.Contains(logs, "service.name: Str(ngf:helloworld:gateway:my-test-svc)")
+		}
+
+		// wait for expected first line to show up
+		Eventually(findTraces, "1m", "5s").Should(BeTrue())
+
+		logs, err := resourceManager.GetPodLogs(collectorNamespace, collectorPodName, &core.PodLogOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(logs).To(ContainSubstring("http.method: Str(GET)"))
+		Expect(logs).To(ContainSubstring("http.target: Str(/hello)"))
+		Expect(logs).To(ContainSubstring("http.target: Str(/world)"))
+		Expect(logs).To(ContainSubstring("testkey1: Str(testval1)"))
+		Expect(logs).To(ContainSubstring("testkey2: Str(testval2)"))
+
+		// verify traces don't exist for helloworld apps
+		Expect(logs).ToNot(ContainSubstring("http.target: Str(/helloworld)"))
+	})
+})
+
+func verifyGatewayClassResolvedRefs() error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+	defer cancel()
+
+	var gc v1.GatewayClass
+	if err := k8sClient.Get(ctx, types.NamespacedName{Name: gatewayClassName}, &gc); err != nil {
```

Comment:
One thing I noticed when I was using Eventually to poll for status updates is that we retry this Get even though the error is not recoverable. My test took a long time to fail, and the error ended up being a typo with the resource name. That's why I chose to use the wait.Poll function instead of Eventually. 

Are we ok with this behavior?

Reply:
Maybe it should be a case by case basis.

Reply:
Well, these cases are pretty much the same...it involves checking the status of a resource. Maybe we can standardize in the future.

Reply:
Ah I see now, I misunderstood earlier. I see benefits both ways. Eventually makes the test itself cleaner I think, but it could make it a bit harder for development.